airflow/jobs.py

```diff
@@ -930,7 +932,20 @@ def _process_task_instances(self, dag, queue, session=None):
             run.dag = dag
             # todo: preferably the integrity check happens at dag collection time
             run.verify_integrity(session=session)
-            run.update_state(session=session)
+            finished_tasks = (
+                session
+                .query(TI.task_id, TI.state
+                       )
+                .filter(
+                    TI.dag_id == run.dag_id,
+                    TI.execution_date == run.execution_date,
```

Comment:
Could we use `run.get_task_instances(state=..., session=session)` here?

Reply:
Yes, I will fix it to be `run.get_task_instances(state=State.finished(), session=session)`